packages/application-shell/src/configure-apollo.ts

```diff
@@ -44,84 +42,37 @@ const httpLink = createHttpLink({
- * as abtract types are used (union and interface). It recommends using the introspection
- * based matcher.
- *
- * The introspection based matcher takes the GraphQL schema as input to more accurately
- * be able to make assumption for caching when looking at a query. The schema is usally
- * queried at build time and injected into the matcher.
- * We can not provide it the schema for mainly two reasons:
- *
- * 1. The CTP schema needs auth and during build not valid auth token is present
- * 2. The schema is different per customer as parts of it are generated
- *
- * Another solution would be to extend the introspection matcher and inject the schema
- * at runtime of the application. However, when passing empty types to the matcher
- * we currently do not loose any of the abilities we use it provides as it only disables
- * caching for union and interface types (https://github.com/apollographql/apollo-client/blob/0166da079da4a9f88eecc7c976ac9d34225f5bf8/packages/apollo-cache-inmemory/src/fragmentMatcher.ts#L150).
- * The InMemoryCache itself uses the matcher to determine if caching is intended (https://github.com/apollographql/apollo-client/blob/0166da079da4a9f88eecc7c976ac9d34225f5bf8/packages/apollo-cache-inmemory/src/writeToStore.ts#L242-L266).
- */
-const introspectionQueryResultData = {
-  __schema: {
-    types: [],
-  },
-};
-const fragmentMatcher = new IntrospectionFragmentMatcher({
-  introspectionQueryResultData,
-});
-
-const typeNamesWithoutIdAsIdentifier = [
-  'Project',
-  'BaseMenu',
-  'NavbarMenu',
-  'Store',
-];
-
-const referenceTypes = ['Reference', 'ChannelReferenceIdentifier'];
-
 export const createApolloClient = () =>
   new ApolloClient({
     link,
+    // https://www.apollographql.com/docs/react/caching/cache-configuration/
     cache: new InMemoryCache({
-      fragmentMatcher,
-      dataIdFromObject: (result) => {
-        const customResult = result as CustomIdGetterObj;
-        if (
-          !customResult.id &&
-          customResult.key &&
-          customResult.__typename &&
-          typeNamesWithoutIdAsIdentifier.includes(customResult.__typename)
-        )
-          return `${customResult.__typename}:${customResult.key}`;
-        // Generally all id's are unique across the platform, so we don't need to
-        // include the type name in the cache key.
-        // However, a reference has the shape { typeId, id } where the id is the
-        // id of the referenced entity. If we didn't prefix ids of References
-        // they would clash with the referenced resource.
-        if (
-          customResult.__typename &&
-          referenceTypes.includes(customResult.__typename)
-        )
-          return `${customResult.__typename}:${customResult.id}`;
-
-        return customResult.id;
+      // https://www.apollographql.com/docs/react/caching/cache-configuration/#generating-unique-identifiers
+      typePolicies: {
```

Comment:
what is missing in the new set up is the reference type.
References can still same id but are of different `typeId`, right?

Reply:
in previous implementation, we made sure that References are unique by leveraging `<typeId>:<id>`.

```js
        if (
          customResult.__typename &&	
          referenceTypes.includes(customResult.__typename)	
        )	
          return `${customResult.__typename}:${customResult.id}`;	

```

This logic is missing now, unless I'm blind

Reply:
Yeah about that. From what I understand with the new cache implementation, one of the braking changes is that fields are [NOT merged by default](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#breaking-cache-changes).

> By default, the `InMemoryCache` no longer merges the fields of two objects unless those objects have the same unique identifier and that identifier is present in both objects. Additionally, the values of fields with the same name are no longer merged recursively by default. You can define a custom `merge` function for a field to handle both of these changes for a particular field.

So I assume (_which needs to be confirmed and tested in our internal apps_) that for such cases we need to provide a merging strategy on our own.

In theory we could implement the `dataIdFromObject` for this specific case (https://www.apollographql.com/docs/react/caching/cache-configuration/#customizing-identifier-generation-globally) but it might be that Apollo removes this functionality in the future, so I'd like to avoid doing that if possible.

Reply:
all right
Then we test this first then, and see if we have to add `dataIdFromObject`. I'd like to avoid that too